webui/react/src/pages/F_ExpList/CompareParallelCoordinates.tsx

```diff
@@ -76,9 +76,13 @@ const CompareParallelCoordinates: React.FC<Props> = ({
 
   const filters: VisualizationFilters = useMemo(
     () => ({
+      batch: undefined,
+      batchMargin: undefined,
       hParams: settings.hParams,
+      maxTrial: undefined,
       metric: settings.metric,
       scale: settings.scale,
+      view: undefined,
```

Comment:
Is there a way to omit undefined/default values? I believe only the non-default values should be passed back and forth.

Reply:
You need to use a partial type.

Reply:
If you only want certain fields to be partial you can do something like:
```typescript
const semiPartial = <P extends t.Props, U extends t.Props>(
   p: P, u: U
): t.UnionC<[t.TypeC<P>, t.PartialC<U>]> => (
   t.union([t.type(p), t.partial(u)])
)
```

Reply:
I tried to work my way using `t.intersection` to construct types with both required and option fields in [this commit](https://github.com/determined-ai/determined/pull/7783/commits/cfe69446c3756c8627a335776e16ee8c111e96a3), but ran into decoding error [here](https://github.com/determined-ai/determined/blob/9b6c6c727a458653a654b669091b7ef120e35847/webui/react/src/stores/userSettings.tsx#L52). Am I on the right track? Do we have to specially decode the intersection type?

Reply:
That looks correct, what was the error you saw?

Reply:
Are you maybe passing `undefined` for a property instead of _not passing_ the property? Typescript treats those as the same since we don't have `exactOptionalPropertyTypes` enabled but they're not actually the same.

Reply:
<img width="1739" alt="Screenshot 2023-09-06 at 10 25 24 AM" src="https://github.com/determined-ai/determined/assets/40620519/d37db420-a517-4868-bf02-00fda5e13262">
This is the error I got while decoding `value` [here](https://github.com/determined-ai/determined/blob/9b6c6c727a458653a654b669091b7ef120e35847/webui/react/src/stores/userSettings.tsx#L52)

Reply:
What's the value that can't be decoded?

Reply:
I believe it's `{scale: "log"}` ? I logged it right before the error